pkg/querier/batch/merge.go

```diff
@@ -66,13 +66,13 @@ func (c *mergeIterator) Reset(size int) *mergeIterator {
 	c.h = c.h[:0]
 	c.batches = c.batches[:0]
 
-	if len(c.its) <= cap(c.batchesBuf) {
+	if size > cap(c.batchesBuf) {
+		c.batchesBuf = make(batchStream, len(c.its))
```

Comment:
Is it better to grow the slice instead of allocating a new one?

Reply:
It would not be the same? go would have to allocate a new one anyway and copy the data over (that's my understanding at least).